src/api/v2/utils.js

```diff
@@ -43,6 +46,31 @@ export function isIp(hostname) {
  * @property {boolean} [isIpAddress]
  */
 
+/**
+ * @type {Set<string> | null}
+ */
+let tldSet = null;
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+
+/**
+ * Get the cached set of top level domains.
+ * @returns {Set<string>}
+ */
+function tlds() {
+  if (!tldSet) {
+    const filePath = path.join(
+      dirname,
+      "..",
+      "..",
+      "..",
+      "conf",
+      "tld-list.json"
+    );
+    tldSet = new Set(JSON.parse(fs.readFileSync(filePath, "utf8")));
```

Comment:
If this errors (due to missing file, or malformed JSON) this will cause all scans to fail - is this what we want, or should we raise an error in sentry but continue scanning (assuming all TLDs are valid)?

Reply:
I think we should halt any scanning if the list is invalid.